src/main/java/com/Crowdsourcing/loot/CrowdsourcingLoot.java

```diff
@@ -0,0 +1,208 @@
+
+	// Hunters' loot sacks
+	private static final String HUNTERS_LOOT_SACK_BASIC = "Hunters' loot sack (basic)";
+	private static final String HUNTERS_LOOT_SACK_ADEPT = "Hunters' loot sack (adept)";
+	private static final String HUNTERS_LOOT_SACK_EXPERT = "Hunters' loot sack (expert)";
+	private static final String HUNTERS_LOOT_SACK_MASTER = "Hunters' loot sack (master)";
+
+	// Clues
+	private static final Pattern CLUE_MESSAGE = Pattern.compile("You have a sneaking suspicion.*");
+
+	// Rogue outfit
+	private static final String ROGUE_MESSAGE = "Your rogue clothing allows you to steal twice as much loot!";
+
+	private HashMap<String, Object> createSkillMap(Skill s)
+	{
+		HashMap<String, Object> h = new HashMap<>();
+		h.put(s.getName(), client.getRealSkillLevel(s));
+		h.put("B" + s.getName(), client.getBoostedSkillLevel(s));
+		return h;
+	}
+
+	private int getRingOfWealth()
+	{
+		ItemContainer equipmentContainer = client.getItemContainer(InventoryID.EQUIPMENT);
+		if (equipmentContainer == null) {
+			return 0;
+		}
+
+		if (equipmentContainer.contains(ItemID.RING_OF_WEALTH) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I))
+		{
+			return 1;
+		}
+
+		if (equipmentContainer.contains(ItemID.RING_OF_WEALTH_1) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_2) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_3) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_4) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_5) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I1) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I2) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I3) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I4) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I5))
+		{
+			return 2;
+		}
+
+		return 0;
+	}
+
+	private HashMap<String, Object> getMetadataForLoot(String name)
+	{
+		HashMap<String, Object> metadata = new HashMap<>();
+
+		// universal metadata
+		metadata.put("ringOfWealth", getRingOfWealth());  // 0=not wearing, 1=wearing uncharged, 2=wearing charged
+
+		for (Map.Entry<Integer, String> entry : VARBITS_CA.entrySet())
+		{
+			int varbitId = entry.getKey();
+			String caTier = entry.getValue();
+			if (client.getVarbitValue(varbitId) == CA_CLAIMED)
+			{
+				metadata.put(caTier, true);
+			}
+		}
+
+		for (Map.Entry<Integer, String> entry : VARBITS_CLUE_WARNINGS.entrySet())
+		{
+			int varbitId = entry.getKey();
+			String clueTier = entry.getValue();
+			if (client.getVarbitValue(varbitId) == CLUE_WARNING_ENABLED)
+			{
+				metadata.put(clueTier, true);
+			}
+		}
+
+		if (name == null)
+		{
+			return metadata;
+		}
+
+		// conditional metadata
+		switch (name)
+		{
+			case HUNTERS_LOOT_SACK_BASIC:
+			case HUNTERS_LOOT_SACK_ADEPT:
+			case HUNTERS_LOOT_SACK_EXPERT:
+			case HUNTERS_LOOT_SACK_MASTER:
+				metadata.putAll(createSkillMap(Skill.HERBLORE));
+				metadata.putAll(createSkillMap(Skill.WOODCUTTING));
+		}
```

Comment:
I would just kill the `createSkillMap` helper here for now to inline it and make this something like
```
if (HUNTERS_LOOT_SACK_BASIC.equals(name) || HUNTERS_LOOT_SACK_ADEPT.equals(name) || ... ) {
        // Add non-boosted and boosted Herblore/Woodcutting levels
	metadata.put("Herblore", client.getRealSkillLevel(Skill.HERBLORE));
	metadata.put("BHerblore", client.getBoostedSkillLevel(Skill.HERBLORE));

	metadata.put("Woodcutting", client.getRealSkillLevel(Skill.WOODCUTTING));
	metadata.put("BWoodcutting", client.getBoostedSkillLevel(Skill.WOODCUTTING));

Reply:
You can either leave this as a helper call for the future or just change to this inline, I don't think it matters too much

Reply:
Ended up removing hunters' loot sacks, and the helper method for now. At the moment I'm not aware of other things beyond the hunters' loot sack that has level-dependent drops, and it doesn't need to be associated with any messages/varbits/etc, so this is better implemented in the loot tracker plugin https://github.com/runelite/runelite/pull/18485